### PR TITLE
Adding name mode section in passing props to route component

### PR DESCRIPTION
**Pull request by suprabhasupi**
Fixes (#2493)

### CODE REVIEW

**Comment by suprabhasupi** [comment]
@posva Is there any issue with this PR. Why its is not getting merged? Please let me if there are changes to be done.

**Comment by posva** [comment]
No, no problems. It's not getting merged yet because I haven't checked :)

**Review by autumnwoodberry — COMMENTED** [review]
This behavior is not really one of the "modes" and the example in the documentation has a comment/example that explains it already:

```
// for routes with named views, you have to define the `props` option for each named view:
{
  path: '/user/:id',
  components: { default: User, sidebar: Sidebar },
  props: { default: true, sidebar: false }
}
```

Perhaps we could remove this comment/example from the code at the top of the page and move it to a new section at the bottom titled "Usage with named views". This way it would be explained as an explicit feature which would be harder to miss. You could even mention that the rules still apply for using the different modes. Maybe something like this:

    ## Usage with named views

    A route with [named views](named-views.html) must have an explicitly defined `props` option for each named view.

    In the following example, the `header` and `default` views will recieve the `id` prop, but the `footer` view will not.

    ```js
    const router = new VueRouter({
      routes: [
        {
          path: '/user/:id',
          components: {
            header: Header,
            default: User,
            footer: Footer
          },
          props: {
            header: true,
            default: true,
          }
        }
      ]
    });
    ```

    The `props` option for each named view can still be defined using any of the available modes:

    ```js
    const router = new VueRouter({
      routes: [
        {
          path: '/user/:id',
          components: {
            header: Header,
            default: User,
            footer: Footer
          },
          props: {
            header: { sticky: true },
            default: route => route.query,
            footer: true,
          }
        }
      ]
    })
    ```

File: docs/guide/essentials/passing-props.md [hunk 1]
```diff
@@ -72,4 +72,22 @@ The URL `/search?q=vue` would pass `{query: 'vue'}` as props to the `SearchUser`
...
+## Name mode
+
+You can pass props to component using their component name.
+
+``` js
+const router = new VueRouter({
+  routes: [
+    {
+      path: '/user-profile',
+      component: {location: Location, userInfo: UserInfo},
```
Comment (tmorehouse) [inline docs/guide/essentials/passing-props.md:84]: Should this be `components`  instead of `component`?

**Comment by posva** [comment]
I think having this documented at https://router.vuejs.org/guide/essentials/passing-props.html as it currently is makes more sense

**Pull request closed without merging**